controllers/argocd/imageUpdater.go

```diff
@@ -0,0 +1,730 @@
+// It does not reconcile/overwrite any fields or information in the configmap itself
+func (r *ReconcileArgoCD) reconcileImageUpdaterConfigMap(cr *argoproj.ArgoCD, desiredConfigMap *corev1.ConfigMap) error {
+	argoutil.AddTrackedByOperatorLabel(&desiredConfigMap.ObjectMeta)
+	return r.reconcileSecretConfigMapHelper(cr, desiredConfigMap)
+}
+
+func (r *ReconcileArgoCD) reconcileImageUpdaterDeployment(cr *argoproj.ArgoCD, sa *corev1.ServiceAccount) error {
+
+	desiredDeployment := newDeploymentWithSuffix(common.ArgoCDImageUpdaterControllerComponent, "controller", cr)
+
+	desiredDeployment.Spec.Strategy = appsv1.DeploymentStrategy{
+		Type: appsv1.RecreateDeploymentStrategyType,
+	}
+
+	imageUpdaterEnv := cr.Spec.ImageUpdater.Env
+	// Let user specify their own environment first
+	imageUpdaterEnv = argoutil.EnvMerge(imageUpdaterEnv, proxyEnvVars(), false)
+
+	podSpec := &desiredDeployment.Spec.Template.Spec
+	podSpec.SecurityContext = &corev1.PodSecurityContext{
+		RunAsNonRoot: boolPtr(true),
+	}
+	AddSeccompProfileForOpenShift(r.Client, podSpec)
+	podSpec.ServiceAccountName = sa.Name
+	podSpec.Volumes = []corev1.Volume{
+		{
+			Name: "tls-certs",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: common.ArgoCDTLSCertsConfigMapName,
+					},
+				},
+			},
+		},
+		{
+			Name: "image-updater-conf",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					Optional: boolPtr(true),
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: ArgocdImageUpdaterConfigCM,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "registries.conf",
+							Path: "registries.conf",
+						},
+						{
+							Key:  "git.commit-message-template",
+							Path: "commit.template",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "ssh-known-hosts",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					Optional: boolPtr(true),
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "argocd-ssh-known-hosts-cm",
+					},
+				},
+			},
+		},
+		{
+			Name: "ssh-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					Optional: boolPtr(true),
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: ArgocdImageUpdaterSSHConfigCM,
+					},
+				},
+			},
+		},
+		{
+			Name: "ssh-signing-key",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: "ssh-git-creds",
+					Optional:   boolPtr(true),
+				},
+			},
+		},
+		{
+			Name: "tmp",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+
+	podSpec.Containers = []corev1.Container{{
+		Command:         []string{"/manager"},
+		Args:            []string{"run"},
+		Image:           argoutil.CombineImageTag(DefaultImageUpdaterImage, DefaultImageUpdaterTag),
```

Comment:
We should make this configurable via env var so that downstream images can be set through the CSV in gitops-operator. 

For ref: https://github.com/argoproj-labs/argocd-operator/blob/ef7197954c1d2ba5b3f733406d294956a78d423a/common/keys.go#L176